src/epac/flatbuffers/pva0_data.py

```diff
@@ -0,0 +1,1002 @@
+
+    Returns:
+        A tuple containing:
+            - The scalar value.
+            - The corresponding FlatBuffers scalar type.
+            - The AnyInner type identifier for the numeric scalar.
+
+    Raises:
+        ValueError: If the data type is not supported for serialisation.
+    """
+    try:
+        return (
+            data.item(),
+            map_dtype_to_scalar_fb[data.dtype],
+            map_dtype_to_any_scalar_enum[data.dtype],
+        )
+    except KeyError:
+        raise ValueError(f"Unsupported scalar dtype: {data.dtype}")
+
+
+def _serialise_string_array(builder: flatbuffers.Builder, data: np.ndarray):
+    """serialises an array of strings into FlatBuffers format.
+
+    Args:
+        builder (flatbuffers.Builder): The FlatBuffers builder used to construct the object.
+        data (np.ndarray): A NumPy array containing string values.
+
+    Returns:
+        A tuple containing:
+            - The FlatBuffers offset for the serialised string array.
+            - The corresponding FlatBuffers StringArray type.
+            - The AnyInner type identifier for a string array.
+    """
+    internal_values_offsets = [builder.CreateString(item) for item in reversed(data)]
+    StringArray.StringArrayStartValueVector(builder, len(data))
+    for start_offset in internal_values_offsets:
+        builder.PrependSOffsetTRelative(start_offset)
+    return builder.EndVector(), StringArray, AnyInner.AnyInner.StringArray
+
+
+def _serialise_numeric_array(builder: flatbuffers.Builder, data: np.ndarray):
+    """serialises an array of numeric values into FlatBuffers format.
+
+    Args:
+        builder (flatbuffers.Builder): The FlatBuffers builder used to construct the object.
+        data (np.ndarray): A NumPy array containing numeric values.
+
+    Returns:
+        A tuple containing:
+            - The FlatBuffers offset for the serialised numeric array.
+            - The corresponding FlatBuffers numeric array type.
+            - The AnyInner type identifier for a numeric array.
+
+    Raises:
+        ValueError: If the data type is not supported for serialisation.
+    """
+    try:
+        return (
+            builder.CreateNumpyVector(data),
+            map_dtype_to_array_fb[data.dtype],
+            map_dtype_to_any_array_enum[data.dtype],
+        )
+    except KeyError:
+        raise ValueError(f"Unsupported scalar dtype: {data.dtype}")
+
+
+def serialise_any(builder: flatbuffers.Builder, data) -> int:
+    """serialises an arbitrary object into a FlatBuffers Any union.
+
+    Args:
+        builder (flatbuffers.Builder): The FlatBuffers builder used to construct the object.
+        data: The object to be serialised, which must be compatible with the Any union.
+
+    Returns:
+        The FlatBuffers offset for the serialised Any object.
+
+    Raises:
+        ValueError: If an attempt is made to serialise an empty array.
+    """
+    data = np.array(data)
```

Comment:
```suggestion
    data = np.asarray(data)
```

I think this avoids a copy if `data` is already an array. I checked and it works when passed objects too.